libraries/lib-audio-devices/DeviceChange.cpp

```diff
@@ -348,8 +348,9 @@ class DeviceChangeListener final : public DeviceChangeInterface
          {
             This->mEnabled = false;
             wxMutexGuiEnter();
-            wxCommandEvent e(EVT_DEVICE_CHANGE);
-            This->mHandler->AddPendingEvent(e);
+            BasicUI::CallAfter([This]{
+               This->mHandler->Publish(DeviceChangeMessage::Change);
```

Comment:
Is there a guarantee that `This` still exists when lambda is invoked?

Reply:
> 

Yes.  The DeviceChangeListener (whatever the OS-specific version of it) is held by a unique_ptr DeviceChangeHandler::mListener.  You can easily verify it is not destroyed until the DeviceChangeHandler is destroyed, and DeviceChangeHandler is only ever created as the base class of DeviceManager, and that class has a static single instance.

Anyway this is compiled only when one of the EXPERIMENTAL flags is switched.